### PR TITLE
Handle error from inet:sockname/1 in mongoose_tcp_listener

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-1660

Proposed changes include:
* Test and case matching

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/127961) / small_tests / f7cd1559425be8b7115ee845f920cb4afdcbdc25
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3695/127961/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3695/127961/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-26_19.48.25/lib.mongooseim.logs/run.2022-06-26_19.48.25/suite.log.html)

File: src/mongoose_tcp_listener.erl [hunk 1]
```diff
@@ -115,34 +115,54 @@ accept_loop(ListenSocket, Module, Opts = #{proxy_protocol := ProxyProtocol}) ->
...
+        {{ok, {DestAddr, DestPort}}, {ok, {SrcAddr, SrcPort}}} ->
             {ok, Socket, #{proxy => false,
                            src_address => SrcAddr,
                            src_port => SrcPort,
                            dest_address => DestAddr,
                            dest_port => DestPort}};
         Other ->
-            Other
+            gen_tcp:close(Socket),
+            {error, simple_reason(Other)}
```
Comment (Premwoik) [inline src/mongoose_tcp_listener.erl:142]: I wonder why CodeCov doesn't recognize any of these error handling cases as covered 🤔

File: src/mongoose_tcp_listener.erl [hunk 2]
```diff
@@ -115,34 +115,54 @@ accept_loop(ListenSocket, Module, Opts = #{proxy_protocol := ProxyProtocol}) ->
...
     {ok, gen_tcp:socket(), connection_details()} | {error, term()}.
-do_accept(ListenSocket, ProxyProtocol) ->
+do_accept(ListenSocket, Opts) ->
     case gen_tcp:accept(ListenSocket) of
-        {ok, Socket} when ProxyProtocol ->
-            read_proxy_header(Socket);
         {ok, Socket} ->
-            {ok, {DestAddr, DestPort}} = inet:sockname(Socket),
-            {ok, {SrcAddr, SrcPort}} = inet:peername(Socket),
+            ?MODULE:read_connection_details(Socket, Opts);
```
Comment (Premwoik) [inline src/mongoose_tcp_listener.erl:123]: Why do we use `?MODULE:` to call the function from the same module? Is this function not always available?

File: src/mongoose_tcp_listener.erl (hunk 1 above)
Comment (arcusfelis) [reply src/mongoose_tcp_listener.erl:142]: because we mock the file under the test, mocked files don't collect coverage info.

Comment (arcusfelis) [reply src/mongoose_tcp_listener.erl:142]: yea, the test runs :)
https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3695/127964/riak_mnesia.24.3.4/big/ct_run.test%4086c1dd277172.2022-06-26_19.48.53/big_tests.tests.tcp_listener_SUITE.logs/run.2022-06-26_20.05.10/suite.log.html

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/128043) / small_tests / a022594cdd23e0c4697de1b86cc6a8b99e9b1f47
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3695/128043/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3695/128043/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-28_09.19.06/lib.mongooseim.logs/run.2022-06-28_09.19.07/suite.log.html)

File: src/mongoose_tcp_listener.erl (hunk 2 above)
Comment (Premwoik) [reply src/mongoose_tcp_listener.erl:123]: Thx

File: src/mongoose_tcp_listener.erl (hunk 1 above)
Comment (Premwoik) [reply src/mongoose_tcp_listener.erl:142]: Ok, thx for the explanation :)

**Pull request merged**